src/main/java/dp/api/dataset/models/Metadata.java

```diff
@@ -231,7 +233,16 @@ public void setQmi(GeneralDetails qmi) {
     }
 
     public String getReleaseDate() {
-        return releaseDate;
+
+        if (this.releaseDate == null) {
+            return "";  // avoid null pointer for empty metadata objects
+        }
+
+        DateTimeFormatter parser = ISODateTimeFormat.dateTimeParser();
```

Comment:
Make both the formatter and parser static final class variables - it's more efficient as it won't have to initialise both object each time the method is called.

Also when a var is `static` and/or `final` the java naming convention is for the names to be all caps i.e. `PARSER` Pedantic I know but if it's worth doing....